### PR TITLE
Adding Example snippets for libseccomp-rs functions in filter_context.rs.

### PR DESCRIPTION
**Pull request by savithry-j**
Signed-off-by: Savithry J <Savithry.Jayaraman@sony.com>

### CODE REVIEW

**Comment by savithry-j** [comment]
Dear Sugimoto san,

Could you please review the PR #133.

Thank you,
Savithry

File: libseccomp/src/lib.rs [hunk 1]
```diff
@@ -616,6 +658,15 @@ impl ScmpFilterContext {
     ///
     /// If this function is called with an invalid filter or the number of syscall
     /// is invalid, an error will be returned.
```
Comment (ManaSugi) [inline libseccomp/src/lib.rs:660]: Could you add a empty line?

File: libseccomp/src/lib.rs [hunk 2]
```diff
@@ -1036,6 +1097,15 @@ impl ScmpFilterContext {
     ///
     /// If this function is called with an invalid filter or an issue is encountered
     /// resetting the filter, an error will be returned.
+    ///
+    ///# Examples
+    ///
+    /// ```
+    /// # use libseccomp::*;
+    /// let mut ctx = ScmpFilterContext::new_filter(ScmpAction::Allow)?;
+    /// ctx.reset(ScmpAction::KillThread)?;
```
Comment (ManaSugi) [inline libseccomp/src/lib.rs:1106]: Could add the following lines?

```rust
let action = ctx.get_act_default()?;
assert_eq!(ScmpAction::KillThread, action);
```

File: libseccomp/src/lib.rs [hunk 3]
```diff
@@ -639,6 +690,16 @@ impl ScmpFilterContext {
...
     /// If this function is called with an invalid filter or an issue is
     /// encountered retrieving the attribute, an error will be returned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use libseccomp::*;
+    /// let mut ctx = ScmpFilterContext::new_filter(ScmpAction::Allow)?;
+    /// ctx.reset(ScmpAction::Allow)?;
+    /// let action = ctx.get_filter_attr(ScmpFilterAttr::ActDefault)?;
```
Comment (ManaSugi) [inline libseccomp/src/lib.rs:700]: Could fix it as follows?

```rust
let action = ctx.get_filter_attr(ScmpFilterAttr::ActDefault)?;
assert_eq!(ScmpAction::Allow, action);
```

**Review by ManaSugi — CHANGES_REQUESTED** [review]
@savithry-j Thank you for the good examples. I have a few comments.

File: libseccomp/src/lib.rs (hunk 1 above)
Comment (savithry-j) [reply libseccomp/src/lib.rs]: Dear @ManaSugi san, Added the empty line.

File: libseccomp/src/lib.rs (hunk 3 above)
Comment (savithry-j) [reply libseccomp/src/lib.rs]: Dear @ManaSugi san,

Modified as below:

```
let mut ctx = ScmpFilterContext::new_filter(ScmpAction::Allow)?;
assert_eq!(ctx.get_filter_attr(ScmpFilterAttr::CtlNnp)?,1);
```

Please review.

**Comment by savithry-j** [comment]
Dear @ManaSugi san,

Resolved merge conflicts and applied fixes for review comments. Could you please re-review.

Thank you,
Savithry

**Review by rusty-snake — COMMENTED** [review]
I would suggest `assert_ne` but otherwise  this looks good to me.

File: libseccomp/src/filter_context.rs [hunk 4]
```diff
@@ -369,6 +422,14 @@ impl ScmpFilterContext {
     ///
     /// If this function is called with an invalid filter or an issue is
     /// encountered retrieving the attribute, an error will be returned.
+    ///
+    /// # Examples
+    /// ```
+    /// # use libseccomp::*;
+    /// let mut ctx = ScmpFilterContext::new_filter(ScmpAction::Allow)?;
+    /// assert!(ctx.get_filter_attr(ScmpFilterAttr::CtlNnp)? != 0);
```
[the remark this answers is not in the document]
Comment (savithry-j) [reply libseccomp/src/filter_context.rs]: Hi @rusty-snake , Thank you for the suggestion. I have committed it. 

Thank you,
Savithry

**Comment by savithry-j** [comment]
Dear Sugimoto san, 

The changes requested in this PR (#133) has been addressed in PR #136 to avoid redundancy. 

Hence I think we can abandon this PR (#133). Could you please let me know if this is fine?

Thank you,
Savithry

**Comment by ManaSugi** [comment]
I'll close this PR because this is coverd by #136.

**Pull request closed without merging**